rest-service/manager_rest/storage/storage_manager.py

```diff
@@ -230,9 +230,12 @@ def _add_tenant_filter(self, query, model_class, all_tenants):
             tenant_ids = [current_tenant.id] if current_tenant else []
 
         # Match any of the applicable tenant ids or if it's a global resource
+        model = (model_class.labeled_model if model_class.is_label
+                 else model_class)
+
         tenant_filter = sql_or(
-            model_class.visibility == VisibilityState.GLOBAL,
-            model_class._tenant_id.in_(tenant_ids)
+            model.visibility == VisibilityState.GLOBAL,
+            model._tenant_id.in_(tenant_ids)
```

Comment:
nice
but the alternative is:
in the label class, do define visibility and tenant id, like so:
```python
@declared_attr
def visibility(cls):
    return cls.labeled_model.visibility
```
then we don't need the if here. It should emit the same/equivalent sql, just different ways of writing it on the python side

which one do you think is better?

Reply:
I think this idea is a lot nicer, will implement it.